training/service.go

```diff
@@ -0,0 +1,366 @@
+
+	if conn, client, err := service.getServiceClient(); err == nil {
+		response, err = client.UpdateModelAccess(ctx, request)
+		log.Infof("Updating model based on response from UpdateModel")
+		if err = service.updateModelDetails(request, response); err != nil {
+			return response, fmt.Errorf("issue with storing Model Id in the Daemon Storage %v", err)
+		}
+		deferConnection(conn)
+	} else {
+		return &ModelDetailsResponse{Status: Status_ERROR}, fmt.Errorf("error in invoking service for Model Training")
+	}
+	return
+}
+
+func (service ModelService) DeleteModel(c context.Context, request *UpdateModelRequest) (response *ModelDetailsResponse,
+	err error) {
+	if request == nil || request.ModelDetailsRequest == nil || request.ModelDetailsRequest.Authorization == nil {
+		return &ModelDetailsResponse{Status: Status_ERROR},
+			fmt.Errorf(" INVALID  REQUEST DETAILS , no Authorization provided  , %v", err)
+	}
+	if err = service.verifySignerForDeleteModel(request.ModelDetailsRequest.Authorization); err != nil {
+		return &ModelDetailsResponse{Status: Status_ERROR},
+			fmt.Errorf(" authentication FAILED , %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*200)
+	defer cancel()
+	if conn, client, err := service.getServiceClient(); err == nil {
+		response, err = client.DeleteModel(ctx, request)
+		log.Infof("Deleting model based on response from DeleteModel")
+		if err = service.deleteModelDetails(request, response); err != nil {
+			return response, fmt.Errorf("issue with deleting Model Id in Storage %v", err)
+		}
+		deferConnection(conn)
+	} else {
+		return &ModelDetailsResponse{Status: Status_ERROR}, fmt.Errorf("error in invoking service for Model Training")
+	}
+
+	return
+}
+
+func (service ModelService) GetModelStatus(c context.Context, request *ModelDetailsRequest) (response *ModelDetailsResponse,
+	err error) {
+	if request == nil || request.Authorization == nil {
+		return &ModelDetailsResponse{Status: Status_ERROR},
+			fmt.Errorf(" INVALID  REQUEST DETAILS , no Authorization provided  , %v", err)
+	}
+	if err = service.verifySignerForGetModelStatus(request.Authorization); err != nil {
+		return &ModelDetailsResponse{Status: Status_ERROR},
+			fmt.Errorf(" authentication FAILED , %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*200)
+	defer cancel()
+
+	if conn, client, err := service.getServiceClient(); err == nil {
+		response, err = client.GetModelStatus(ctx, request)
+		log.Infof("Updating modelG based on response from UpdateModel")
+		if err = service.updateModelDetailsForStatus(request, response); err != nil {
+			return response, fmt.Errorf("issue with storing Model Id in the Daemon Storage %v", err)
+		}
+		deferConnection(conn)
+	} else {
+		return &ModelDetailsResponse{Status: Status_ERROR}, fmt.Errorf("error in invoking service for Model Training")
+	}
+	return
+}
+
+//message used to sign is of the form ("__create_model", mpe_address, current_block_number)
```

Comment:
Lets not have different methods, instead have the caller pass in the prefix which should be a constant

Reply:
makes sense will do